requirements.txt

```diff
@@ -0,0 +1,3 @@
+pyOpenSSL
+requests
+boto3
```

Comment:
The "miniupnpc" library is an optional requirement too, used when you're behind a firewall (like most Raspberry Pis')

Which tool/app uses requirements.txt ?

Reply:
It's much easier for users, instead of installing the dependencies the can use:

"Requirements files" are files containing a list of items to be installed using pip install like `pip install -r requirements.txt`

Refer to [the documentation here](https://pip.pypa.io/en/stable/user_guide/#requirements-files). This makes setup easier for users. We should also version pin, which I'll add later.

`sam build` also uses this file, it will package all the dependencies required by the Lambda function.

Reply:
Cool, the requirements look useful, I see a lot of other Python projects have a requirements file.

So in AWS the config.json file is essentially in the ParameterStore. I was wondering about state.json, which basically stores the bunq installation. That's required because a bunq api token is only valid for 4 hours. Where should the lambda store its state?

Reply:
It stored on the scratch space until the function is 'recycled' which happens frequently. When that happens bunq2ynab starts 'fresh' and the `state.json` is recreated.

Reply:
> When that happens bunq2ynab starts 'fresh' and the `state.json` is recreated.

I think you can only create new installations for 4 hours after you create the API token. The installation is basically the private key + installation token that's stored in state.json. So I'd be surprised if that worked for a long time.

It's only the session token that can be endlessly refreshed.

Reply:
If that is true, we would need to store those in SSM as well. That is how I'm doing it today and would not be a problem.

However, sometimes the install is no-longer valid and we somehow need to detect and refresh the state json. Could you point me to the code where you detect and handle that?

Reply:
I've never seen an installation token become invalid just because it's old. I think you have only 4 hours after creating an API key to create an installation token.

As far as I know the only thing that becomes invalid over time is the session token. That's detected at https://github.com/wesselt/bunq2ynab/blob/config/lib/bunq.py#L189.  The code will request a new session token and ask the State class to store it.